src/shogun/features/CombinedFeatures.cpp

```diff
@@ -162,6 +162,16 @@ void CombinedFeatures::init()
 {
 	SG_ADD(&num_vec, "num_vec", "Number of vectors.");
 	SG_ADD(&feature_array, "feature_array", "Feature array.");
+	add_callback_function("feature_array", [this](){
+		int32_t n=feature_array.back()->get_num_vectors();
+		if (get_num_vectors()>0 && n!=get_num_vectors())
+		{
+			feature_array.pop_back();
+			error("Number of feature vectors does not match (expected {}, "
+					"obj has {})", get_num_vectors(), n);
+		}
+		num_vec=n;
+	});
```

Comment:
@karlnapf these callbacks are becoming really awkward. The issue is if something hasn't been set then we need to reset to the old state. In this case this is possible because I can pop the last element of the vector (which would be invalid here). The problem is that callbacks are only triggered after updating the parameter. What do you think about adding a customisable constraint where you can just pass lambda that checks the new parameter value before replacing the old parameter value?

Reply:
this is a design flaw, having the check in here. Algorithms should check this when training imo.

having said that, there should definitely be checks that are done *before* changing, just like the constraints. Can't you have a custom constraint here btw?

Reply:
Well, you can but you need to create a new struct which is quite verbose, so I was thinking I could add a customisable constraint where you just have a lambda.
```cpp
SG_ADD(&array, "feature_array", SG_CONSTRAINT([this](const auto& new_element){
  if (get_num_vectors()>0 && n!=get_num_vectors())
  {
	error("Number of feature vectors does not match (expected {}, obj has {})", get_num_vectors(), n);
  }
}))
```

I am not sure why the check there is flawed though? Would you not want to know about an issue as soon as possible, ie. before training?

Reply:
And then that would have to also have a callback to update `num_vec` (we could have that in the constraint but then we are mixing functionality)

Reply:
num_vec should just be removed and always be queried from the first feature (if exists), and then everying can be matched against that.